src/oscar/apps/order/abstract_models.py

```diff
@@ -223,24 +223,23 @@ def shipping_tax(self):
 
     @property
     def shipping_status(self):
-        events = self.shipping_events.all()
+        events = self.shipping_events.order_by('-date_created').all()
         if not len(events):
             return ''
 
         # Collect all events by event-type
-        map = {}
+        event_map = OrderedDict()
         for event in events:
             event_name = event.event_type.name
-            if event_name not in map:
-                map[event_name] = []
-            map[event_name] = list(chain(map[event_name],
-                                         event.line_quantities.all()))
+            if event_name not in event_map:
+                event_map[event_name] = []
+            event_map[event_name].extend(list(event.line_quantities.all()))
 
         # Determine last complete event
         status = _("In progress")
-        for event_name, event_line_quantities in map.items():
+        for event_name, event_line_quantities in event_map.items():
             if self._is_event_complete(event_line_quantities):
-                status = event_name
+                return event_name
```

Comment:
Since the queryset orders on the `date_created` descending the last complete status is returned.

I'll add some extra tests for this